### PR TITLE
add last_checked, reserved_at and execution_duration to cc background:queue:status

### PR DESCRIPTION
**Pull request by mrow4a**
BEFORE:
```
root@93b6da3d89fa: /var/www/owncloud # occ background:queue:status
+--------+----------------------------------------------------+---------------------------+---------------+
| Job ID | Job                                                | Last Run                  | Job Arguments |
+--------+----------------------------------------------------+---------------------------+---------------+
| 1      | OCA\Files\BackgroundJob\ScanFiles                  | 1970-01-01T00:00:00+00:00 |               |
| 2      | OCA\Files\BackgroundJob\DeleteOrphanedItems        | 1970-01-01T00:00:00+00:00 |               |
| 3      | OCA\Files\BackgroundJob\CleanupFileLocks           | 1970-01-01T00:00:00+00:00 |               |
| 4      | OCA\Files\BackgroundJob\CleanupPersistentFileLocks | 1970-01-01T00:00:00+00:00 |               |
| 5      | OCA\DAV\CardDAV\SyncJob                            | 1970-01-01T00:00:00+00:00 |               |
| 6      | OCA\DAV\BackgroundJob\CleanProperties              | 1970-01-01T00:00:00+00:00 |               |
| 7      | OCA\Activity\BackgroundJob\EmailNotification       | 1970-01-01T00:00:00+00:00 |               |
| 8      | OCA\Activity\BackgroundJob\ExpireActivities        | 1970-01-01T00:00:00+00:00 |               |
| 9      | OCA\Federation\SyncJob                             | 1970-01-01T00:00:00+00:00 |               |
| 10     | OCA\Files_Sharing\DeleteOrphanedSharesJob          | 1970-01-01T00:00:00+00:00 |               |
| 11     | OCA\Files_Sharing\ExpireSharesJob                  | 1970-01-01T00:00:00+00:00 |               |
| 12     | OCA\Files_Sharing\External\ScanExternalSharesJob   | 1970-01-01T00:00:00+00:00 |               |
| 13     | OCA\Files_Trashbin\BackgroundJob\ExpireTrash       | 1970-01-01T00:00:00+00:00 |               |
| 14     | OCA\Files_Versions\BackgroundJob\ExpireVersions    | 1970-01-01T00:00:00+00:00 |               |
| 15     | OCA\Market\CheckUpdateBackgroundJob                | 1970-01-01T00:00:00+00:00 |               |
| 16     | OCA\UpdateNotification\Notification\BackgroundJob  | 1970-01-01T00:00:00+00:00 |               |
| 17     | OC\Authentication\Token\DefaultTokenCleanupJob     | 1970-01-01T00:00:00+00:00 |               |
+--------+----------------------------------------------------+---------------------------+---------------+
```

AFTER:
```
+--------+----------------------------------------------------+---------------+---------------------------+---------------------------+---------------------------+------------------------+
| Job ID | Job                                                | Job Arguments | Last Run                  | Last Checked              | Reserved At               | Execution Duration (s) |
+--------+----------------------------------------------------+---------------+---------------------------+---------------------------+---------------------------+------------------------+
| 1      | OCA\Files\BackgroundJob\ScanFiles                  |               | N/A                       | 2022-05-31T19:17:05+00:00 | 2022-05-31T19:17:05+00:00 | N/A                    |
| 2      | OCA\Files\BackgroundJob\DeleteOrphanedItems        |               | N/A                       | 2022-05-31T19:18:07+00:00 | 2022-05-31T19:18:07+00:00 | N/A                    |
| 3      | OCA\Files\BackgroundJob\CleanupFileLocks           |               | N/A                       | 2022-05-31T19:19:05+00:00 | 2022-05-31T19:19:05+00:00 | N/A                    |
| 4      | OCA\Files\BackgroundJob\CleanupPersistentFileLocks |               | N/A                       | 2022-05-31T19:20:04+00:00 | 2022-05-31T19:20:04+00:00 | N/A                    |
| 5      | OCA\DAV\CardDAV\SyncJob                            |               | N/A                       | 2022-05-31T19:21:06+00:00 | 2022-05-31T19:21:06+00:00 | N/A                    |
| 6      | OCA\DAV\BackgroundJob\CleanProperties              |               | N/A                       | 2022-05-31T19:22:05+00:00 | 2022-05-31T19:22:05+00:00 | N/A                    |
| 7      | OCA\Activity\BackgroundJob\EmailNotification       |               | N/A                       | 2022-05-31T19:23:05+00:00 | 2022-05-31T19:23:05+00:00 | N/A                    |
| 8      | OCA\Activity\BackgroundJob\ExpireActivities        |               | 2022-05-31T19:24:05+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 9      | OCA\Federation\SyncJob                             |               | 2022-05-31T19:24:05+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 10     | OCA\Files_Sharing\DeleteOrphanedSharesJob          |               | 2022-05-31T19:40:06+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 11     | OCA\Files_Sharing\ExpireSharesJob                  |               | 2022-05-31T19:24:05+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 12     | OCA\Files_Sharing\External\ScanExternalSharesJob   |               | 2022-05-31T19:35:06+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 13     | OCA\Files_Trashbin\BackgroundJob\ExpireTrash       |               | 2022-05-31T19:35:06+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 14     | OCA\Files_Versions\BackgroundJob\ExpireVersions    |               | 2022-05-31T19:24:05+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 15     | OCA\Market\CheckUpdateBackgroundJob                |               | 2022-05-31T19:24:05+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 2                      |
| 16     | OCA\UpdateNotification\Notification\BackgroundJob  |               | 2022-05-31T19:24:07+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
| 17     | OC\Authentication\Token\DefaultTokenCleanupJob     |               | 2022-05-31T19:40:06+00:00 | 2022-05-31T19:40:06+00:00 | N/A                       | 0                      |
+--------+----------------------------------------------------+---------------+---------------------------+---------------------------+---------------------------+------------------------+
```
## Description
Complete occ command output with all available job attributes

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/5205

## How Has This Been Tested?
- executing `occ background:queue:status` in all possible combinations

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: lib/public/BackgroundJob/IJob.php [hunk 1]
```diff
@@ -58,6 +58,24 @@ public function setLastRun($lastRun);
 	 */
 	public function setArgument($argument);
 
+	/**
+	 * @param int $lastChecked
+	 * @since 10.11.0
+	 */
+	public function setLastChecked($lastChecked);
```
Comment (DeepDiver1975) [inline lib/public/BackgroundJob/IJob.php:65]: type declaration please

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
see comments

File: lib/private/BackgroundJob/Job.php [hunk 2]
```diff
@@ -117,4 +144,16 @@ public function getLastRun() {
 	public function getArgument() {
 		return $this->argument;
 	}
+
+	public function getLastChecked() {
```
Comment (DeepDiver1975) [inline lib/private/BackgroundJob/Job.php:148]: please add return types - we are on php 7.3+ which allows this - THX

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
...

File: lib/private/BackgroundJob/Job.php (hunk 2 above)
Comment (mrow4a) [reply lib/private/BackgroundJob/Job.php]: done

File: lib/public/BackgroundJob/IJob.php (hunk 1 above)
Comment (mrow4a) [reply lib/public/BackgroundJob/IJob.php]: done

**Comment by mmattel** [comment]
@mrow4a you may need to rebase to check if CI is happy. We are going to have a 10.11 release, this would be a perfect time to finish and merge this PR.

**Comment by mrow4a** [comment]
@DeepDiver1975 completed gaps in unit tests complaining about types. Ready for re-review so that we can fit it in 10.11 release

P.S. It was very good idea to add types, it has shown many incomplete unit tests and forced to make sure logic is correct

**Review by DeepDiver1975 — APPROVED** [review]

**Review by phil-davis — COMMENTED** [review]
Needs a changelog entry

**Comment by phil-davis** [comment]
I will rebase, squash and adjust the changelog...

**Review by phil-davis — APPROVED** [review]

**Pull request merged**